### PR TITLE
pmlib: new package

### PR DESCRIPTION
**Pull request by Hironori-Yamaji**
This PR adds (new) support to build and install the `pmlib` package.

### CODE REVIEW

File: var/spack/repos/builtin/packages/pmlib/package.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+                '-DCMAKE_CXX_FLAGS=-fopenmp',
+                '-DCMAKE_Fortran_FLAGS=-fopenmp -cpp',
+            ]
+            args = args + gcc_args
+
+        if '%fj' in spec:
+            fj_args = [
+                '-DCMAKE_TOOLCHAIN_FILE=./cmake/Toolchain_fx100.cmake',
+                '-Denable_OPENMP=no',
+                '-Dwith_MPI=yes',
```
Comment (scheibelp) [inline var/spack/repos/builtin/packages/pmlib/package.py]: Should this be `-Dwith_MPI=no` if the `~mpi` variant is disabled?

File: var/spack/repos/builtin/packages/pmlib/package.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    variant('mpi', default=False, description='Activate MPI support')
+
+    phases = ['edit', 'cmake', 'build', 'install']
+
+    depends_on('mpi', when='+mpi')
+
+    def edit(self, spec, prefix):
+        if '%fj' in self.spec:
+            toolchain = FileFilter('./cmake/Toolchain_fx100.cmake')
+            toolchain.filter('mpifccpx', spec['mpi'].mpicc)
```
Comment (scheibelp) [inline var/spack/repos/builtin/packages/pmlib/package.py]: Is `cmake/Toolchain_fx100.cmake` used when compiling with GCC? If so, then I think these edits should always be applied.

File: var/spack/repos/builtin/packages/pmlib/package.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+       including hybrid(OpenMP & MPI) code. In addition, PAPI interface allows
+       us to access the information of build-in hardware counter."""
+
+    homepage = "https://github.com/avr-aics-riken/PMlib"
+    git      = "https://github.com/avr-aics-riken/PMlib.git"
+
+    version('master', branch='master')
+    version('6.4.1', commit='0a35f5bec8c12e532e5a1bdac8c32c659fd3ee11')
+
+    variant('mpi', default=False, description='Activate MPI support')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/pmlib/package.py]: This should probably default to True: https://spack.readthedocs.io/en/latest/packaging_guide.html#variant-names

File: var/spack/repos/builtin/packages/pmlib/package.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+    phases = ['edit', 'cmake', 'build', 'install']
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/pmlib/package.py]: I would just rename `edit` to `patch` and keep the default phases.

File: var/spack/repos/builtin/packages/pmlib/package.py (hunk 1 above)
Comment (Hironori-Yamaji) [reply var/spack/repos/builtin/packages/pmlib/package.py]: Thank you for your reply.
Yes. I added variants and `if` statements, so we could select that variants is able or disable.

File: var/spack/repos/builtin/packages/pmlib/package.py (hunk 2 above)
Comment (Hironori-Yamaji) [reply var/spack/repos/builtin/packages/pmlib/package.py]: `cmake/Toolchain_fx100.cmake` is not used when compiling with GCC.
Therefore, this `if` statement is necessary, but I changed `edit` phase to `patch`.

File: var/spack/repos/builtin/packages/pmlib/package.py (hunk 3 above)
Comment (Hironori-Yamaji) [reply var/spack/repos/builtin/packages/pmlib/package.py]: Thank you for your help.
I changed `default=False` to `default=True`.

File: var/spack/repos/builtin/packages/pmlib/package.py (hunk 4 above)
Comment (Hironori-Yamaji) [reply var/spack/repos/builtin/packages/pmlib/package.py]: I changed `edit` phase to `patch`.

File: var/spack/repos/builtin/packages/pmlib/fix_compiler_options.patch [hunk 5]
```diff
@@ -23,7 +23,7 @@ diff -urN PMlib-org/cmake/Toolchain_fx100.cmake PMlib-fix/cmake/Toolchain_fx100.
 -  CMAKE_FORCE_CXX_COMPILER(mpiFCCpx GNU)
 -  CMAKE_FORCE_Fortran_COMPILER(mpifrtpx GNU)
 +  CMAKE_FORCE_C_COMPILER(mpicc GNU)
-+  CMAKE_FORCE_CXX_COMPILER(mpicc GNU)
++  CMAKE_FORCE_CXX_COMPILER(mpicxx GNU)
```
Comment (Hironori-Yamaji) [inline var/spack/repos/builtin/packages/pmlib/fix_compiler_options.patch]: Sorry for the lack of confirmation. 
There was no need to correct this problem in my environment.
But, I correct this part to be more flexibility.

**Pull request merged**